### DISCUSSION TITLE
Read Selection

### DISCUSSION THREAD

**Discussion by Randalix** [gokcehan/lf#892] [Q&A] [answered]
Can I read a selection from a file and have lf display the visual clue for a selection?

**Comment by DusanLesan** [comment, marked as answer]
I do not understand the requirements. Can you describe your use case?

**Reply by Randalix** [reply]
sure!
To illustrate I take sxiv as an example: While viewing images I decide to move some of them. For this purpose I write out a file containing the paths to the images. Now it would be great if I could use that file to make a preselection in lf.
Other example git: 
I have a list of  uncommited files. I would like to have a little commit wrapper which preselects those files and letting me make the final selection.
Does those examples help.

**Reply by DusanLesan** [reply]
I already do something similar for nsxiv:
```
cmd open ${{
	case $(file --mime-type "$f" -b) in
		image/*) lf -remote "send $id openimages" ;;
		*/octet-stream) dmenumount "$f" ;;
		*/zip) lf -remote "send $id mountarchive" ;;
		text/*) $EDITOR $fx ;;
		*) for f in $fx; do setsid $OPENER $f >/dev/null 2>&1 & done ;;
	esac
}}

cmd openimages ${{
	devour nsxiv -aiop 2>/dev/null <<< "$fx" |
	(lf -remote "send $id unselect"
	while read -r file; do
		lf -remote "send $id select \"$file\""
		lf -remote "send toggle"
	done)
}}
```
Here I use ability of sxiv to mark files, and write out selection on exit(`-o` option). After exiting I first use `lf -remote "send unselect"` to unselect all files I selected to open. Then I use remote ability of `lf` to select and toggle all selected files one by one like: `lf -remote "send select /home/dusan/.config"; lf -remote "send toggle"`. Devour is just providing swallowing and is not needed for this flow.

Like this, you can loop through any list of files and make the selections(and toggle)

**Reply by DusanLesan** [reply]
It looks like you can toggle with file name right away without the need to select first like: `lf -remote "send toggle /home/dusan/.config"`. You can still select one of them if you want to move focus at that path

**Reply by DusanLesan** [reply]
Did the sxiv example work out for you?

**Comment by Randalix** [comment]
Sorry for the late reply. This is perfect. Thank you very much!

**Reply by DusanLesan** [reply]
No problem @Randalix. You can mark the post as answered so this discussion does not sit as `Unanswered`
